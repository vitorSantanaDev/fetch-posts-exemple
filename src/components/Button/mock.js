export const buttonProps = {
  text: 'Load More Posts',
  onClick: jest.fn(),
  disabled: false,
};
