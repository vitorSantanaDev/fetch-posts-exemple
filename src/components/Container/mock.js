import React from 'react';

export const propsContainer = {
  children: <h1>Children</h1>,
};
