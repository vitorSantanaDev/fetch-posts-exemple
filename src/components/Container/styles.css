.container {
  padding: 30px;
  margin: 0 auto;
  min-height: 100vh;
  max-width: 1300px;
  background-color: #ededed;
}