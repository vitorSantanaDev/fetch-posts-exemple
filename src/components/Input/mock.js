export const propsInput = {
  value: '',
  onChange: jest.fn(),
};
