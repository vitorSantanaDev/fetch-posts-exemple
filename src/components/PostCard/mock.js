export const postCardProps = {
  title: 'title one',
  body: 'Body one',
  cover: 'img/img.png',
};
