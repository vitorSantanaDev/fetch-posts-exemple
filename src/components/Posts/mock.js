export const postsProps = [
  { id: 1, title: 'Title Post one', body: 'Body text Post one', cover: '/image/image.png' },
  { id: 2, title: 'Title Post two', body: 'Body text Post two', cover: '/image/image.png' },
  { id: 3, title: 'Title Post three', body: 'Body text Post three', cover: '/image/image.png' },
];
