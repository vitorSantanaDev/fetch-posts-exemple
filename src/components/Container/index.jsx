import React from 'react';
import './styles.css';
import Types from 'prop-types';

const Container = ({ children }) => {
  return <div className="container">{children}</div>;
};

Container.propTypes = {
  children: Types.node.isRequired,
};

export default Container;
