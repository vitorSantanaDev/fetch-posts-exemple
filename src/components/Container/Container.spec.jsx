import { render, screen } from '@testing-library/react';
import Container from '.';
import { propsContainer } from './mock';

describe('<Container />', () => {
  it('should render Container', () => {
    render(<Container>{propsContainer.children}</Container>);
    const h1 = screen.getByRole('heading', { name: /children/i });
    expect(h1).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Container>{propsContainer.children}</Container>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
