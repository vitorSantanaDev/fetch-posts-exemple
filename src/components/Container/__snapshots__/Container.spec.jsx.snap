// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should match snapshot 1`] = `
<div
  class="container"
>
  <h1>
    Children
  </h1>
</div>
`;
